src/listeners/message-listeners/fdp-storage.listener.ts

```diff
@@ -0,0 +1,65 @@
+import BackgroundAction from '../../constants/background-actions.enum'
+import { isFdpStorageRequest } from '../../messaging/message.asserts'
+import { FdpStorageRequest } from '../../model/internal-messages.model'
+import { DappPermissions } from '../../model/storage/dapps.model'
+import { Dialog } from '../../services/dialog.service'
+import { callFdpStorageMethod } from '../../services/fdp-storage/fdp-storage-access'
+import { dappUrlToId } from '../../services/fdp-storage/fdp-storage.utils'
+import { SessionFdpStorageProvider } from '../../services/fdp-storage/session-fdp-storage.provider'
+import { Storage } from '../../services/storage/storage.service'
+import { SwarmExtension } from '../../swarm-api/swarm-extension'
+import { createMessageHandler } from './message-handler'
+
+const fdpStorageProvider = new SessionFdpStorageProvider()
+const dialogs = new Dialog()
+const storage = new Storage()
+
+async function handleFdpStorageRequest(
+  { accessor, parameters }: FdpStorageRequest,
+  sender: chrome.runtime.MessageSender,
+): Promise<unknown> {
+  try {
+    const [fdp, { extensionId }] = await Promise.all([fdpStorageProvider.getService(), storage.getSwarm()])
+
+    if (!fdp) {
+      throw new Error('Blossom: User is not logged in.')
+    }
+
+    const swarmExtension = new SwarmExtension(extensionId)
+
+    const { beeApiUrl } = await swarmExtension.beeAddress()
+
+    const dappId = dappUrlToId(sender.url, beeApiUrl)
+
+    const dapp = await storage.getDapp(dappId)
+
+    if (!dapp) {
+      // TODO specify permissions
+      const confirmed = await dialogs.ask('DIALOG_CREATE_POD', { name: dappId })
+
+      if (!confirmed) {
+        throw new Error('Blossom: Access denied')
+      }
+
+      // TODO support different types of permission
+      await storage.updateDapp(dappId, { permissions: [DappPermissions.CREATE_POD] })
+    }
+
+    const [property, method] = accessor.split('.')
```

Comment:
Yes, that's correct. But all fdp-storage methods are accessed in form of `fdp.someProperty.someMethod(...)`. I don't see any case where methods are nested more.

Reply:
but it can happen really easily in the future.

Reply:
If it happens we will have to update our logic as well. Because we cannot just allow all methods to be invoked. We must have some kind of a list of allowed methods. And if there might be more levels of objects, then the property variable should be changed to array. I'll make that fix.

Reply:
please create an issue about it in order to not forget this realization.

Reply:
A list of allowed methods is already implemented, I'm saying just that we will have to update it whenever there is a breaking change in the fdp-storage. But I'll create an issue for supporting nested objects in fdp-storage and we can implement it later, because we currently don't need it.